shared/src/main/scala/amf/plugins/document/vocabularies/parser/instances/DialectInstanceParser.scala

```diff
@@ -334,36 +334,105 @@ class DialectInstanceParser(val root: Root)(implicit override val ctx: DialectIn
+
+        val computedNodeId: String = mapping.idTemplate.option() match {
+          case Some(idTemplate) =>
+
+            // Check if it also has an $id overwriting the base URL
+            base = refMap.key("$id") match {
+              case Some(baseId) if baseId.value.tagType == YType.Str && (baseId.value.as[String].contains("://") ||  baseId.value.as[String].startsWith("//")) =>
+                Some(baseId.value.as[String]) // I will overwrite wthatever might be in the ID template
+              case Some(_)  =>
+                ctx.eh.violation(DialectError, id, "External link with map values and relative $id, $id must be absolute if provided with a link map $id")
+                None
+              case _ =>
+                None
+            }
+
+            // compute a relative template
+            var template = if (idTemplate.contains("://")) {
+              idTemplate.split("://").last
+            } else {
+              idTemplate
+            }
+            if (template.startsWith("//")) {
+              template = template.drop(2)
+            }
+
+            // now generate the final template
+            var separator = ""
+            template = if (base.isEmpty) { // no idTemplate -> use all the template
+              idTemplate
+            } else if (template.contains("/")) { // idTemplate base/path/to/endpoint(#optionalFragment), I need a relative path
+              val parts = template.split("/")
+              separator = "/"
+              parts.drop(1).mkString("/")
+            } else if (template.contains("#")) {  // idTemplate baseWithoutPath#fragment , I need a relative path
+              val parts = template.split("#")
+              separator = "#"
+              parts.drop(1).mkString("#")
+            } else { // default case, Is this an error?
+              template
+            }
+            val finalPath = templateNodeId(id, refMap, template)
+            if (base.isEmpty) {
+              finalPath
+            } else {
+              base.getOrElse(root.location) ++ separator ++ finalPath
+            }
+
+          case None             =>
+            ctx.eh.violation(DialectError, id, "External link with map values and no idTemplate defined for the node in the dialect")
+            "/anon-node-error"
+        }
+
+        // Now we actually parse the provided properties for the node
+        val linkReference: DialectDomainElement = DialectDomainElement(Annotations(refMap)).withDefinedBy(mapping).withId(computedNodeId)
+        linkReference.withInstanceTypes(Seq(mapping.nodetypeMapping.option(), Some(mapping.id)).collect { case Some(t) => t })
+        linkReference.withIsExternalLink(true) // this is a linkReference
+
+        // explicit $id
+        if (base.isDefined) {
+          linkReference.annotations += CustomId(base.get)
+        }
+
+        mapping.propertiesMapping().foreach { propertyMapping =>
```

Comment:
@antoniogarrote why do we need to parse property for external links? Wouldn't it be sufficient to parse properties for template ids?